internal/controller/kube/kube.go

```diff
@@ -27,7 +28,48 @@ func GetVolume(ctx context.Context, kube client.Client, volumeName, ns string) (
 	return obj, err
 }
 
-// IsVolumeDeleted - checks if a volume is ready
+func GetNic(ctx context.Context, kube client.Client, name, ns string) (*v1alpha1.Nic, error) {
+	obj := &v1alpha1.Nic{}
+	if err := kube.Get(ctx, types.NamespacedName{
+		Namespace: ns,
+		Name:      name,
+	}, obj); err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+// // GetVolume - returns a volume object from kubernetes
+// func ListVolumesFromIndex(ctx context.Context, kube client.Client, volumeName, ns string) (*v1alpha1.Volume, error) {
+// 	obj := &v1alpha1.Volume{}
+// 	err := kube.Get(ctx, types.NamespacedName{
+// 		Namespace: ns,
+// 		Name:      volumeName,
+// 	}, obj)
+//
+// 	return obj, err
+// }
+
+func ListResourceFromLabelIndex(ctx context.Context, kube client.Client, resType string, index int, obj client.ObjectList) error {
+	if err := kube.List(ctx, obj, client.MatchingLabels{
+		fmt.Sprintf("ionoscloud.com/serverset-%s-index", resType): strconv.Itoa(index),
```

Comment:
this should live within serverset pkg, due to label relation `"ionoscloud.com/serverset-%s-index", resType"`,